tests/common/dualtor/dual_tor_utils.py

```diff
@@ -87,4 +93,244 @@ def get_t1_ptf_ports(dut, tbinfo):
+        tbinfo (dict): Testbed info from the tbinfo fixture.
+
+    Yields:
+        function: A function for shutting down fanout interfaces connected to specified upper_tor interfaces
+    """
+    down_intfs = []
+
+    def shutdown(dut_intfs=None):
+        _down_intfs = shutdown_fanout_tor_intfs(upper_tor_host, upper_tor_fanouthosts, tbinfo, dut_intfs)
+        for fanout_host, fanout_intf in _down_intfs:
+            down_intfs.append((fanout_host, fanout_intf))
+
+    yield shutdown
+
+    for fanout_host, fanout_intf in down_intfs:
+        fanout_host.no_shutdown(fanout_intf)
+
+
+@pytest.fixture
+def shutdown_fanout_lower_tor_intfs(lower_tor_host, lower_tor_fanouthosts, tbinfo):
+    """Fixture for shutting down fanout interfaces connected to specified lower_tor interfaces.
+
+    Args:
+        lower_tor_host (object): Host object for lower_tor.
+        lower_tor_fanouthosts (dict): Key is fanout hostname, value is fanout host object.
+        tbinfo (dict): Testbed info from the tbinfo fixture.
+
+    Yields:
+        function: A function for shutting down fanout interfaces connected to specified lower_tor interfaces
+    """
+    down_intfs = []
+
+    def shutdown(dut_intfs=None):
+        _down_intfs = shutdown_fanout_tor_intfs(lower_tor_host, lower_tor_fanouthosts, tbinfo, dut_intfs)
+        for fanout_host, fanout_intf in _down_intfs:
+            down_intfs.append((fanout_host, fanout_intf))
+
+    yield shutdown
+
+    for fanout_host, fanout_intf in down_intfs:
+        fanout_host.no_shutdown(fanout_intf)
+
+
+def shutdown_t1_tor_intfs(tor_host, nbrhosts, tbinfo, vm_names=None):
+    """Function for shutting down specified T1 VMs' interfaces that are connected to the tor_host.
+
+    Args:
+        tor_host (object): Host object for the ToR DUT.
+        nbrhosts (dict): Dict returned by the nbrhosts fixture.
+        tbinfo (dict): Testbed info from the tbinfo fixture.
+        vm_names (list, optional): List of VM names, for example: ['ARISTA01T1', 'ARISTA02T1']. All the interfaces
+            connected to tor_host on the specified VMs will be shutdown. If vm_names is None, shutdown will be performed
+            on all the T1 VMs of tor_host.
+            Defaults to None.
+
+    Returns:
+        list of tuple: Return a list of tuple. Each tuple has two items. The first item is the host object for VM.
+            The second item is the VM interface that has been shutdown. The returned list makes it easy to recover
+            the interfaces.
+    """
+    down_intfs = []
+
+    tor_index = tbinfo['duts_map'][tor_host.hostname]
+
+    if not vm_names:
+        target_vms = nbrhosts
+    else:
+        target_vms = {}
+        for vm_name in vm_names:
+            if vm_name in nbrhosts:
+                target_vms[vm_name] = nbrhosts[vm_name]
+            else:
+                logger.error('Unknown vm_name: "{}"'.format(vm_name))
+
+    for vm_name in natsorted(target_vms.keys()):
```

Comment:
Is there a reason to run this loop in a sorted VM list?

Reply:
There is no special reason. This is just to make the test behavior more predictable when the sequence does not matter.